### PR TITLE
MGDAPI-412: Ability to read addon parameters

### PR DESCRIPTION
**Pull request by sergioifg94**
# Description

> Link to Jira: https://issues.redhat.com/browse/MGDAPI-412

Create `addon` package to include addon related logic, and add logic to retrieve addon parameters.
Add logic to set the CIRD value from the addon parameters.

## Verification steps

On a cluster

1. Prepare the cluster and ensure the `cloud-resources-aws-strategies` ConfigMap doesn't exist
    ```sh
    make cluster/prepare/local
    oc delete configmaps cloud-resources-aws-strategies -n redhat-rhmi-operator
    ```
1. Create the parameters secret
    ```sh
    oc create secret generic addon-managed-api-service-parameters -n redhat-rhmi-operator \
      --from-literal=cidr-range=10.1.0.0/23 \
      --from-literal=notification-email=test-customer@redhat.com
    ```
1. Install MGDAPI locally
    ```sh
    INSTALLATION_TYPE=managed-api make code/run
    ```
1. Verify the CIRD parameter is reconciled
    1. Wait for the cloud-resources stage to complete
    2. Verify that the ConfigMap has been created and the CIDR block has been included
        ```
        oc get configmaps cloud-resources-aws-strategies -n redhat-rhmi-operator -o json | jq .data | jq -r "._network" | jq     ".production.createStrategy.CidrBlock" 
    
        "10.1.0.0/23"
        ```
1. Verify the email address parameter is reconciled
    1. Check that the `.spec.alertingEmailAddress` is `test-customer@redhat.com`
    2. Check that the address is included in the `BUAndCustomer` and `SREAndCustomer` receivers in the Alert Manager configuration:
        1. Log in the alert manager
            >
            > You can find the route with:
            > ```sh
            > oc get route alertmanager-route -n redhat-rhmi-middleware-monitoring-operator
            >
        1. Go to `Status -> Config` and check the `to` fields for the receivers

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: pkg/products/cloudresources/reconciler.go [hunk 1]
```diff
@@ -340,3 +347,58 @@ func overrideStrategyConfig(resourceType string, tier string, croStrategyConfig
 
 	return nil
 }
+
+// reconcileCIDRValue sets the CIDR value in the ConfigMap from the addon
```
Comment (pmccarthy) [inline pkg/products/cloudresources/reconciler.go:351]: If possible we should also look to read in the notifications email address that will be added as part of https://gitlab.cee.redhat.com/service/managed-tenants/-/merge_requests/310/diffs#diff-content-ebae93872f6cae079f57ba32f871b9eaf446340c

**Comment by jjaferson** [comment]
/lgtm

Verified on an OSD cluster

**Comment by philbrookes** [comment]
/retest
cluster not created properly

**Comment by sergioifg94** [comment]
/hold
Modifying this to allow a list of multiple emails in the `notification-email` parameter

**Comment by KevFan** [comment]
Verified  :+1: 

Screenshot from 2020-11-09 15-17-29

Screenshot from 2020-11-09 15-18-41

/lgtm

**Pull request merged**